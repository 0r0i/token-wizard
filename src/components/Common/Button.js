import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
