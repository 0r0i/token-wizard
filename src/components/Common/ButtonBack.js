import React, { Component } from 'react'

export class ButtonBack extends Component {
  /**
   * Render method for ButtonBack component
   * @returns {*}
   */
  render() {
    const { onClick } = this.props

    return (
      <button onClick={onClick} type="button" className={`sw-ButtonBack`}>
        Back
      </button>
    )
  }
}
