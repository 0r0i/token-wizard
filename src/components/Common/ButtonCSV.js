import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
