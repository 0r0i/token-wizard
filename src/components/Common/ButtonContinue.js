import React from 'react'

export const ButtonContinue = ({ disabled = false, type, onClick, extraClassName, buttonText = 'Continue' }) => (
  <button
    className={`sw-ButtonContinue ${extraClassName ? extraClassName : ''}`}
    disabled={disabled}
    onClick={onClick}
    type={type}
  >
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
