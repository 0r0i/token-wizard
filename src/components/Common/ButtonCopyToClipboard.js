import React from 'react'
import { copy, showClipboardCopyToast } from '../../utils/copy'

const elementClass = 'sw-ButtonCopyToClipboard'

copy(`.${elementClass}`)

export const ButtonCopyToClipboard = ({ disabled, extraClassName = '', value, title = '' }) => (
  <button
    className={`${elementClass} ${extraClassName}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
    onClick={() => {
      showClipboardCopyToast(title)
    }}
  />
)
