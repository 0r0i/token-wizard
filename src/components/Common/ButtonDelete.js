import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
