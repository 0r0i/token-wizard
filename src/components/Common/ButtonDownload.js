import React, { Component } from 'react'

export class ButtonDownload extends Component {
  render() {
    const { disabled, onClick, extraClass } = this.props
    return (
      <button
        className={`sw-ButtonDownload ${extraClass ? extraClass : ''}`}
        disabled={disabled}
        onClick={onClick}
        type="button"
      >
        Download
      </button>
    )
  }
}
