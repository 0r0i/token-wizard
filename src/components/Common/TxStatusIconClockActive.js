import React from 'react'

export const TxStatusIconClockActive = () => (
  <svg id="animatedClock" xmlns="http://www.w3.org/2000/svg" width="19" height="19" viewBox="0 0 19 19">
    <defs>
      <style
        type="text/css"
        dangerouslySetInnerHTML={{
          __html: `
            @keyframes clock-animation {
              0% {
                transform: rotate(0);
                transform-origin: center center;
              }
              100% {
                transform: rotate(360deg);
                transform-origin: center center;
              }
            }
            #animatedClockMinutes {
              animation: clock-animation 5s steps(120, end) infinite both;
            }
          `
        }}
      />
    </defs>
    <path
      fill="#ffffff"
      d="M9.419 0C4.225 0 0 4.225 0 9.419c0 5.193 4.225 9.418 9.419 9.418 5.193 0 9.418-4.225 9.418-9.418C18.837 4.225 14.612 0 9.42 0zm0 16.833c-4.089 0-7.415-3.326-7.415-7.414 0-4.089 3.326-7.415 7.415-7.415 4.088 0 7.414 3.326 7.414 7.415 0 4.088-3.326 7.414-7.414 7.414z"
    />
    <g stroke="#ffffff" strokeLinecap="round" strokeWidth="1.6">
      <line id="animatedClockMinutes" x1="9.3" y1="4" x2="9.3" y2="10" />
      <line id="animatedClockHours" x1="9.8" x2="14.2" y1="9.5" y2="9.5" />
    </g>
  </svg>
)
