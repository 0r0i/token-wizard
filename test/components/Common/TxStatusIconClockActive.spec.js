import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'
import { TxStatusIconClockActive } from '../../../src/components/Common/TxStatusIconClockActive'

configure({ adapter: new Adapter() })

describe(`TxStatusIconClockActive`, () => {
  it(`should be a <svg> tag`, () => {
    const wrapper = shallow(<TxStatusIconClockActive />)

    expect(wrapper.type()).toBe('svg')
  })

  it(`should render TxStatusIconClockActive component`, () => {
    // Given
    const component = render.create(<TxStatusIconClockActive />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
