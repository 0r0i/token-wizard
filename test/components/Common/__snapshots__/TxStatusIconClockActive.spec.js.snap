// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TxStatusIconClockActive should render TxStatusIconClockActive component 1`] = `
<svg
  height="19"
  id="animatedClock"
  viewBox="0 0 19 19"
  width="19"
  xmlns="http://www.w3.org/2000/svg"
>
  <defs>
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": "
            @keyframes clock-animation {
              0% {
                transform: rotate(0);
                transform-origin: center center;
              }
              100% {
                transform: rotate(360deg);
                transform-origin: center center;
              }
            }
            #animatedClockMinutes {
              animation: clock-animation 5s steps(120, end) infinite both;
            }
          ",
        }
      }
      type="text/css"
    />
  </defs>
  <path
    d="M9.419 0C4.225 0 0 4.225 0 9.419c0 5.193 4.225 9.418 9.419 9.418 5.193 0 9.418-4.225 9.418-9.418C18.837 4.225 14.612 0 9.42 0zm0 16.833c-4.089 0-7.415-3.326-7.415-7.414 0-4.089 3.326-7.415 7.415-7.415 4.088 0 7.414 3.326 7.414 7.415 0 4.088-3.326 7.414-7.414 7.414z"
    fill="#ffffff"
  />
  <g
    stroke="#ffffff"
    strokeLinecap="round"
    strokeWidth="1.6"
  >
    <line
      id="animatedClockMinutes"
      x1="9.3"
      x2="9.3"
      y1="4"
      y2="10"
    />
    <line
      id="animatedClockHours"
      x1="9.8"
      x2="14.2"
      y1="9.5"
      y2="9.5"
    />
  </g>
</svg>
`;
